x-pack/plugins/observability/public/hooks/slo/mix_kql_with_tags.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SearchState } from '../../pages/slos/hooks/use_url_search_state';
+
+export function mixKqlWithTags(kqlQuery: string, tags: SearchState['tags']) {
+  if (!tags) {
+    return kqlQuery;
+  }
+  const tagsKqlIncluded = tags.included?.join(' or ') || '';
+  const excludedTagsKql = tags.excluded?.join(' or ') || '';
+
+  let tagsQuery = '';
+  if (tagsKqlIncluded && excludedTagsKql) {
+    tagsQuery = `slo.tags: (${excludedTagsKql}) and not slo.tags: (${tagsKqlIncluded})`;
+  }
+  if (!excludedTagsKql && tagsKqlIncluded) {
+    tagsQuery = `slo.tags: (${tagsKqlIncluded})`;
+  }
+  if (!tagsKqlIncluded && excludedTagsKql) {
+    tagsQuery = `not slo.tags: (${excludedTagsKql})`;
+  }
+
+  if (!kqlQuery) {
+    return tagsQuery;
+  }
+
+  if (tagsQuery) {
+    return `${kqlQuery} and ${tagsQuery}`;
+  } else {
+    return kqlQuery;
+  }
```

Comment:
Can't we simplify to this?
```suggestion
  let queryParts = [];
  if (!!kqlQuery) {
    queryParts.push(kqlQuery);
  }
  if (!!includedKqlTags) {
    queryParts.push(`slo.tags: (${includedKqlTags})`);
  }
  if (!!excludedKqlTags) {
    queryParts.push(`not slo.tags: (${excludedKqlTags})`);
  }
  return queryParts.join(" and ");
```

Reply:
Looks good! I ran the tests with the suggested changes and all green!